Source/PowerMeter/PowerMeter.cs

```diff
@@ -0,0 +1,134 @@
+            /// </summary>
+            public bool Averaging;
+
+            /// <summary>
+            /// Specifies the frequency in Hz of the power to be measured since this frequency is not automatically determined.
+            /// </summary>
+            public double Frequency;
+
+            /// <summary>
+            /// Specifies the ContAv Aperture in seconds.
+            /// </summary>
+            public double ApertureTime;
+
+            /// <summary>
+            /// Specifies the filter bandwidth.
+            /// </summary>
+            public int Count;
+
+            /// <summary>
+            /// Returns the struct with default values set.
+            /// </summary>
+            public static CommonConfiguration GetDefault()
+            {
+                var commonConfiguration = new CommonConfiguration
+                {
+                    Channel_Name = "0",
+                    Frequency = 2E9,
+                    Units = 1,
+
+                    Measurement_Mode = ni568xConstants.ContinuousMode,
+                    Trigger_Source = ni568xConstants.Immediate,
+                    Slope = ni568xConstants.Positive,
+                    Trigger_Level = -20,
+                    Averaging = true,
+                    ApertureTime = 0.02,
+                    Count = 4
+                };
+                return commonConfiguration;
+            }
+        }
+        #endregion
+
+        #region Configuration
+        /// <summary>
+        /// Configures common settings for power meter.
+        /// </summary>
+        /// <param name="sensor">Specifies a reference to the instrument.</param>
+        /// <param name="commonConfiguration">Specifies the settings for power meter.</param>
+        public static void ConfigureCommon(ni568x sensor, CommonConfiguration commonConfiguration)
+        {
+            sensor.ConfigureAcquisitionMode(commonConfiguration.Channel_Name, commonConfiguration.Measurement_Mode);
+            sensor.ConfigureUnits(commonConfiguration.Units);
+            sensor.ConfigureTriggerSource(commonConfiguration.Trigger_Source);
+
+            if (commonConfiguration.Trigger_Source == ni568xConstants.Internal)
+            {
+                sensor.ConfigureInternalTrigger(commonConfiguration.Channel_Name, commonConfiguration.Slope);
+                sensor.ConfigureInternalTriggerLevel(commonConfiguration.Trigger_Level);
+            }
+
+            sensor.ConfigureRangeAutoEnabled(commonConfiguration.Channel_Name, true);
+            sensor.ConfigureCorrectionFrequency(commonConfiguration.Channel_Name, commonConfiguration.Frequency);
+            if (commonConfiguration.Averaging)
+            {
+                sensor.ConfigureAveragingAutoEnabled(commonConfiguration.Channel_Name, commonConfiguration.Averaging);
+            }
+            else
+            {
+                sensor.ConfigureAveragingCount(commonConfiguration.Channel_Name, commonConfiguration.Count);
+            }
```

Comment:
I don't understand this logic. If this "Averaging" variable is set to true, we turn on auto averaging, otherwise we set the number of averages.

Is "Averaging" supposed to be AutoAveragingEnabled? That would make sense with the current implementation.

Reply:
Is "Averaging" supposed to be AutoAveragingEnabled? That's correct. I have updated the comments for "Averaging".